R/target_sda.R

```diff
@@ -131,20 +131,39 @@ target_sda <- function(data,
   check_crucial_names(co2_intensity_scenario, crucial_scenario)
   walk_(crucial_scenario, ~ check_no_value_is_missing(co2_intensity_scenario, .x))
 
+  region_isos <- region_isos %>%
+    filter(source %in% unique(co2_intensity_scenario$scenario_source))
+
+  if (identical(nrow(region_isos), 0L)) {
+    warn("Found no matching regions for input scenario", class = "no_match")
+    return(empty_target_sda_output())
+  }
+
+  ald <- ald %>%
+    mutate(plant_location = tolower(.data$plant_location)) %>%
```

Comment:
nit: Why are the country isos lower case? Technically, they are upper case, so if someone uses data with proper isos they don't match.

Reply:
The contents of the data in `r2dii.data` is always formatted lowercase, so this line ensures that the `ald` input, which by default has `plant_location` uppercase, is formatted to lowercase, so that the join matches. 

I see the point though, if the user uses a different `region_isos` file, we would need to also force them lowercase. Hmm, I will make a new issue for this.